src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/TraceLogging/EventPayload.cs

```diff
@@ -42,7 +43,7 @@ internal EventPayload(List<string> payloadNames, List<object?> payloadValues)
                     throw new System.ArgumentNullException(nameof(key));
 
                 int position = 0;
-                foreach (string name in m_names)
+                foreach (string name in CollectionsMarshal.AsSpan(m_names))
```

Comment:
You may need to revert all of the changes under System/Diagnostics/Tracing, as these files are shared with corefx where a netstandard library is built to enable EventSource downlevel.

Reply:
Also, we should use `CollectionsMarshal.AsSpan` only in performance critical places. None of the places in this file look performance critical.